jsregexp.c

```diff
@@ -36,6 +38,17 @@ struct regexp {
   uint32_t last_index;
 };
 
+struct jsstring {
+  bool is_wide_char;
+  uint32_t len;
+  char* bstr; // base string passed in
+  uint32_t* indices;
+  union {
+      uint8_t* str8; /* 8 bit strings will get an extra null terminator */
+      uint16_t* str16;
+  } u;
+};
```

Comment:
An anonymous union could be used (from what I can tell quickjs needs at least `c11`).

Reply:
I’d prefer to keep it c99 for my own use case.